packages/cli/src/models/files/ZosNetworkFile.ts

```diff
@@ -0,0 +1,365 @@
+  }
+
+  public getDependency(name: string): any | null {
+    if (!this.data.dependencies) return null;
+    return this.data.dependencies[name] || {};
+  }
+
+  public hasDependency(name: string): boolean {
+    return !_.isEmpty(this.getDependency(name));
+  }
+
+  public hasDependencies(): boolean {
+    return !_.isEmpty(this.dependencies);
+  }
+
+  public getProxies({ package: packageName, contract, address }: {package?: string, contract?: any, address?: string} = {}): any[] {
+    if (_.isEmpty(this.data.proxies)) return [];
+    const allProxies = _.flatMap(this.data.proxies || {}, (proxiesList, fullname) => (
+      _.map(proxiesList, (proxyInfo) => ({
+        ...fromContractFullName(fullname),
+        ...proxyInfo
+      }))
+    ));
+    return _.filter(allProxies, (proxy) => (
+      (!packageName || proxy.package === packageName) &&
+      (!contract || proxy.contract === contract) &&
+      (!address || proxy.address === address)
+    ));
+  }
+
+  // TS-TODO: return type can probably be typed to Proxy.ts from lib?
+  public getProxy(address: string): any {
+    const allProxies = this.getProxies();
+    return _.find(allProxies, { address });
+  }
+
+  public contract(alias: string): any {
+    return this.data.contracts[alias];
+  }
+
+  public contractAliasesMissingFromPackage(): any[] {
+    return _.difference(this.contractAliases, this.packageFile.contractAliases);
+  }
+
+  public isCurrentVersion(version: string): boolean {
+    return this.version === version;
+  }
+
+  public hasContract(alias: string): boolean {
+    return !_.isEmpty(this.contract(alias));
+  }
+
+  public hasContracts(): boolean {
+    return !_.isEmpty(this.data.contracts);
+  }
+
+  public hasProxies(filter: any = {}): boolean {
+    return !_.isEmpty(this.getProxies(filter));
+  }
+
+  public hasMatchingVersion(): boolean {
+    return this.packageFile.isCurrentVersion(this.version);
+  }
+
+  public dependenciesNamesMissingFromPackage(): any[] {
+    return _.difference(this.dependenciesNames, this.packageFile.dependenciesNames);
+  }
+
+  public dependencyHasCustomDeploy(name: string): boolean {
+    const dep = this.getDependency(name);
+    return dep && dep.customDeploy;
+  }
+
+  public dependencySatisfiesVersionRequirement(name: string): boolean {
+    const dep = this.getDependency(name);
+    return dep && this.packageFile.dependencyMatches(name, dep.version);
+  }
+
+  public dependencyHasMatchingCustomDeploy(name: string): boolean {
+    return this.dependencyHasCustomDeploy(name) && this.dependencySatisfiesVersionRequirement(name);
+  }
+
+  public hasSameBytecode(alias: string, klass: any): boolean {
+    const contract = this.contract(alias) || this.solidityLib(alias);
+    if (contract) {
+      const localBytecode = contract.localBytecodeHash;
+      const currentBytecode = bytecodeDigest(klass.bytecode);
+      return currentBytecode === localBytecode;
+    }
+  }
+
+  set version(version: string) {
```

Comment:
If we can, let's try to define all return types (there are also some other methods below without them)